go/logic/migrator.go

```diff
@@ -1279,30 +1279,31 @@ func (this *Migrator) executeWriteFuncs() error {
 					return err
 				}
 			}
-		default:
-			{
+		case copyRowsFunc := <-this.copyRowsQueue:
+		priority:
+			for {
 				select {
-				case copyRowsFunc := <-this.copyRowsQueue:
+				case eventStruct := <-this.applyEventsQueue:
 					{
-						copyRowsStartTime := time.Now()
-						// Retries are handled within the copyRowsFunc
-						if err := copyRowsFunc(); err != nil {
-							return this.migrationContext.Log.Errore(err)
-						}
-						if niceRatio := this.migrationContext.GetNiceRatio(); niceRatio > 0 {
-							copyRowsDuration := time.Since(copyRowsStartTime)
-							sleepTimeNanosecondFloat64 := niceRatio * float64(copyRowsDuration.Nanoseconds())
-							sleepTime := time.Duration(int64(sleepTimeNanosecondFloat64)) * time.Nanosecond
-							time.Sleep(sleepTime)
+						if err := this.onApplyEventStruct(eventStruct); err != nil {
+							return err
 						}
 					}
 				default:
-					{
-						// Hmmmmm... nothing in the queue; no events, but also no row copy.
-						// This is possible upon load. Let's just sleep it over.
-						this.migrationContext.Log.Debugf("Getting nothing in the write queue. Sleeping...")
-						time.Sleep(time.Second)
-					}
+					break priority
```

Comment:
@debug-LiXiwen I am new to using labeled-`break`/`continue`. Sorry for the noob question

I assume `break priority` causes the program to jump-back to `priority:` [on line 1283](https://github.com/github/gh-ost/pull/1124/files#diff-bd6a81585ccffb0f0feef9d5b3d1c542432f90c29a366a7bfb1e96531d36cbbaR1283). Will [the `for` loop on line 1284](https://github.com/github/gh-ost/pull/1124/files#diff-bd6a81585ccffb0f0feef9d5b3d1c542432f90c29a366a7bfb1e96531d36cbbaR1284) on execute on the 2nd+ iteration?

I believe we want to empty `this.applyEventsQueue` once and continue to row copy, so I want to confirm

Reply:
It’s valid to terminate outer loops
labeled-break will jump labeled code and  move to the next clause